src/codeql.ts

```diff
@@ -131,3 +110,93 @@ async function downloadDatabase(
     }
   }
 }
+
+// Calls `bqrs info` and returns compatible query kinds for the given bqrs file
+async function getCompatibleQueryKinds(
+  codeql: string,
+  bqrs: string
+): Promise<string[]> {
+  const bqrsInfoOutput = await getExecOutput(codeql, [
+    "bqrs",
+    "info",
+    "--format=json",
+    bqrs,
+  ]);
+  if (bqrsInfoOutput.exitCode !== 0) {
+    throw new Error(
+      `Unable to run codeql bqrs info. Exit code: ${bqrsInfoOutput.exitCode}`
+    );
+  }
+  return JSON.parse(bqrsInfoOutput.stdout)["compatible-query-kinds"];
+}
+
+// Generates results.csv from the given bqrs file
+async function outputCsv(codeql: string, bqrs: string): Promise<string> {
+  const csv = path.join("results", "results.csv");
+  await exec(codeql, [
+    "bqrs",
+    "decode",
+    "--format=csv",
+    `--output=${csv}`,
+    bqrs,
+  ]);
+  return csv;
+}
+
+// Generates results.md from the given bqrs file
+async function outputMd(
+  codeql: string,
+  bqrs: string,
+  nwo: string,
+  compatibleQueryKinds: string[]
+): Promise<string> {
+  const json = path.join("results", "results.json");
+  await exec(codeql, [
+    "bqrs",
+    "decode",
+    "--format=json",
+    `--output=${json}`,
+    "--entities=all",
+    bqrs,
+  ]);
+
+  const sourceLocationPrefix = JSON.parse(
+    (await getExecOutput(codeql, ["resolve", "database", "db"])).stdout
+  ).sourceLocationPrefix;
+
+  // This will load the whole result set into memory. Given that we just ran a
+  // query, we probably have quite a lot of memory available. However, at some
+  // point this is likely to break down. We could then look at using a streaming
+  // parser such as http://oboejs.com/
+  const jsonResults = JSON.parse(fs.readFileSync(json, "utf8"));
+
+  const md = path.join("results", "results.md");
+  const s = fs.createWriteStream(md, {
+    encoding: "utf8",
+  });
+
+  await interpret(
+    s,
+    jsonResults,
+    nwo,
+    compatibleQueryKinds,
+    sourceLocationPrefix,
+    "HEAD"
```

Comment:
This looks like it might conflict with #154

Reply:
Yes, but hopefully won't be too hard to resolve. @shati-patel, just let me know if it's hard to resolve.

Reply:
Should be fine! Thanks for the heads-up 🌷